hooks/read-package-hook/src/createVersionsOverrider.ts

```diff
@@ -61,25 +58,28 @@ interface VersionOverride extends VersionOverrideBase {
   localTarget?: LocalTarget
 }
 
-type VersionOverrideWithParent = VersionOverride & Required<Pick<VersionOverride, 'parentPkg'>>
+interface VersionOverrideWithParent extends VersionOverride {
+  parentPkg: PackageSelector
+}
 
 function overrideDepsOfPkg (
   { manifest, dir }: { manifest: PackageManifest, dir: string | undefined },
   versionOverrides: VersionOverrideWithParent[],
   genericVersionOverrides: VersionOverride[]
 ): void {
-  if (manifest.dependencies != null) overrideDeps(versionOverrides, genericVersionOverrides, manifest.dependencies, dir)
-  if (manifest.optionalDependencies != null) overrideDeps(versionOverrides, genericVersionOverrides, manifest.optionalDependencies, dir)
-  if (manifest.devDependencies != null) overrideDeps(versionOverrides, genericVersionOverrides, manifest.devDependencies, dir)
-  if (manifest.peerDependencies != null) overrideDeps(versionOverrides, genericVersionOverrides, manifest.peerDependencies, dir)
+  const { dependencies, optionalDependencies, devDependencies, peerDependencies } = manifest
+  for (const deps of [dependencies, optionalDependencies, devDependencies, peerDependencies]) {
+    overrideDeps(versionOverrides, genericVersionOverrides, deps, dir)
```

Comment:
i'd still prefer this
```suggestion
  if (deps) {
    overrideDeps(versionOverrides, genericVersionOverrides, deps, dir)
  }
```

When a function is executed it should do something. Otherwise the name of the function isn't really correct.

Reply:
https://github.com/pnpm/pnpm/pull/8573/commits/dfaa2392a7d058e69b8748c07fef5f97955b634d